backends-common/postgres/src/main/java/org/apache/james/backends/postgres/utils/JamesPostgresConnectionFactory.java

```diff
@@ -21,9 +21,11 @@
 
 import org.apache.james.core.Domain;
 
-import io.r2dbc.spi.Connection;
+import io.r2dbc.postgresql.api.PostgresqlConnection;
 import reactor.core.publisher.Mono;
 
 public interface JamesPostgresConnectionFactory {
-    Mono<? extends Connection> getConnection(Domain domain);
+    Mono<? extends PostgresqlConnection> getConnection(Domain domain);
```

Comment:
Is `Mono<PostgresqlConnection> getConnection(Domain domain);` not enough?

Reply:
switch back to connection